plugin-server/src/main/ingestion-queues/batch-processing/each-batch-buffer.ts

```diff
@@ -2,15 +2,14 @@ import { EachBatchPayload, KafkaMessage } from 'kafkajs'
 
 import { runInstrumentedFunction } from '../../utils'
 import { KafkaQueue } from '../kafka-queue'
-
-class DelayProcessing extends Error {}
+import { formPluginEvent } from './each-batch-ingestion'
 
 export async function eachMessageBuffer(
     message: KafkaMessage,
     resolveOffset: EachBatchPayload['resolveOffset'],
     queue: KafkaQueue
 ): Promise<void> {
-    const bufferEvent = JSON.parse(message.value!.toString())
+    const bufferEvent = formPluginEvent(message)
```

Comment:
Is `formPluginEvent` idempotent? It has already been called once at this point in main topic consumption, not sure this is a safe refactor.

Reply:
Didn't notice that `emitToBufferStep` emits the event in its "formed" form, reverted this.